### PR TITLE
CIRC-815 fix issue with removing expired sessions

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Approach 
- fix issue with removing expired sessions which have nonexistent loanId or patronId;
- add tests;

Resolves: [CIRC-815](https://issues.folio.org/browse/CIRC-815) 

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionRepository.java [hunk 1]
```diff
@@ -177,9 +177,13 @@ private boolean isPatronActionTypeSpecified(PatronActionType actionType) {
   private MultipleRecords<PatronSessionRecord> setUsersForLoans(
     MultipleRecords<PatronSessionRecord> records, Map<String, User> usersMap) {
 
-    return records.mapRecords(sessionRecord -> sessionRecord
-      .withLoan(sessionRecord.getLoan().withUser(
-        usersMap.get(sessionRecord.getPatronId().toString()))));
+    return records.mapRecords(sessionRecord -> {
+      if (sessionRecord.getLoan() != null) {
+        return sessionRecord.withLoan(sessionRecord.getLoan().withUser(
+          usersMap.get(sessionRecord.getPatronId().toString())));
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionRepository.java:183]: Possible NPE? sessionRecord.getPatronId()

File: src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionService.java [hunk 2]
```diff
@@ -125,16 +125,23 @@ public PatronActionSessionService(
   private CompletableFuture<Result<MultipleRecords<PatronSessionRecord>>> sendNoticesForAllUsers(
     MultipleRecords<PatronSessionRecord> records) {
 
-    if (records == null || records.isEmpty()) {
+    if (records.isEmpty()) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionService.java:128]: Null check?

Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionService.java]: added

File: src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionRepository.java (hunk 1 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionRepository.java:184]: fixed

**Pull request merged**